crates/composefs-boot/src/uki.rs

```diff
@@ -78,21 +78,59 @@ pub enum UkiError {
     NoName,
 }
 
-/// Extracts a text section from a UKI PE file by name.
+/// Extracts a text section from a UKI PE file by name and validates it as UTF-8.
 ///
-/// Parses the PE file format to locate and extract the contents of a named
-/// text section (e.g., ".osrel", ".cmdline"). Returns None if the PE format
-/// is invalid, Some(Ok) with the text content if found and valid UTF-8, or
-/// Some(Err) with the specific error.
+/// This is a convenience wrapper around [`get_section`] that additionally validates
+/// the section contents as valid UTF-8 text.
+///
+/// # Arguments
+///
+/// * `image` - The complete UKI image as a byte slice
+/// * `section_name` - Name of the PE section to extract (e.g., ".osrel", ".cmdline")
+///
+/// # Returns
+///
+/// * `None` - If the PE format is invalid or cannot be parsed
+/// * `Some(Ok(&str))` - If the section is found and contains valid UTF-8
+/// * `Some(Err(UkiError))` - If the section is missing or contains invalid UTF-8
+pub fn get_text_section<'a>(
+    image: &'a [u8],
+    section_name: &'static str,
+) -> Option<Result<&'a str, UkiError>> {
+    get_section(image, section_name).map(|res| {
```

Comment:
can't you just use `?.and_then` for that instead of the `map`?  `None` on an `Option` works the same way as `Err`...